recovery/recovery.go

```diff
@@ -0,0 +1,282 @@
+		return "", errors.New(e_zero_content)
+	}
+
+	defer resp.Body.Close()
+
+	// read body
+	body := new(bytes.Buffer)
+	body.ReadFrom(resp.Body)
+
+	// if !r.isNewContent(hashPath, body.Bytes()) {
+	// 	return "", errors.New(e_content_not_changed)
+	// }
+	return body.String(), nil
+}
+
+/////////////////////////////////////////////////////////////
+func getWDPath() string {
+	cwd, err := os.Getwd()
+	if err != nil {
+		panic(err)
+	}
+	return cwd + "/boyar_recovery/"
+}
+
+/////////////////////////////////////////////////////////////
+func (r *Recovery) tick() {
+	logger.Info("Recovery tick")
+	r.tickCount += 1
+	r.lastTick = time.Now()
+
+	// read json
+	jsnTxt, err := r.readUrl(r.config.Url) //, getWDPath())
+	if err != nil {
+		r.lastReadErr = err.Error()
+		logger.Error(err.Error())
+		return
+	}
+	var result Instructions
+	//var result map[string]interface{}
+	err = json.Unmarshal([]byte(jsnTxt), &result)
+	if err != nil {
+		r.lastReadErr = err.Error()
+		logger.Error(err.Error())
+	}
+	//no scripts to run
+	scriptArr := result.Shell.Run
+	if len(scriptArr) == 0 {
+		r.lastReadErr = "json run array came empty"
+		logger.Error(r.lastReadErr)
+		return
+	}
+	// no executable for bash
+	if result.Shell.Bin == "" {
+		r.lastReadErr = "bin for exec was not specified in json"
+		logger.Error(r.lastReadErr)
+		return
+	}
+	// clean last output for status
+	r.lastOutput = ""
+
+	// execute all scripts serial
+	for _, url := range scriptArr {
+		// read script
+		script, err := r.readUrl(url) //, getWDPath())
+		if err != nil {
+			r.lastReadErr = err.Error()
+			logger.Error(err.Error())
+		} else {
+			r.runScript(result.Shell.Bin, script)
+		}
+	}
+
+}
+func (r *Recovery) runScript(bin, script string) {
+	// reset error
+	r.lastReadErr = ""
+
+	// no prefix
+	if len(script) < 4 {
+		r.lastReadErr = "script length < 4"
+		logger.Error(r.lastReadErr)
+		return
+	}
+
+	// #!/ prefix check
+	if script[:3] != "#!/" {
+		r.lastReadErr = e_no_bash_prefix
+		logger.Error(r.lastReadErr)
+		return
+	}
+
+	// execute
+	logger.Info("Recovery about to execute script")
+	logger.Info("------------------------------")
+	logger.Info(script)
+	logger.Info("------------------------------")
+
+	out, err := execBashScript(bin, script)
+	r.lastExec = time.Now()
+	if len(out) > 0 {
+		logger.Info("output")
+		logger.Info(out)
+		r.lastOutput += out
+	} else {
+		logger.Error("exec Error")
+		logger.Error(err.Error())
+		r.lastOutput = "ERROR: " + err.Error()
+	}
+	logger.Info("------------------------------")
+}
+
+func (r *Recovery) Status() interface{} {
+	return map[string]interface{}{
+		"IntervalMinute": r.config.IntervalMinute,
+		"Url":            r.config.Url,
+		"tickCount":      r.tickCount,
+		"lastTick":       r.lastTick,
+		"lastExec":       r.lastExec,
+		"lastHash":       r.lastHash,
+		"lastOutput":     r.lastOutput,
+		"lastReadError":  r.lastReadErr,
+	}
+}
+
+func execBashScript(bin, script string) (string, error) {
+	shell := os.Getenv("SHELL")
```

Comment:
Ideally don't make any assumptions it's even a shell script, so I would remove all this